modules/tinymce/src/core/main/ts/api/EditorUpload.ts

```diff
@@ -135,7 +136,19 @@ const EditorUpload = function (editor: Editor): EditorUpload {
             blobCache.removeByUri(image.src);
             replaceImageUriInView(image, uploadInfo.url);
           } else if (uploadInfo.error) {
-            ErrorReporter.uploadError(editor, uploadInfo.error);
+            if (uploadInfo.error.options.remove) {
+              if (Rtc.isRtc(editor)) {
+                // To be replaced by RTC API to mirror DOM changes when such is implemented.
+                console.log('Could not erase image due to RTC plugin'); // eslint-disable-line no-console
+              } else {
+                editor.undoManager.transact(() => {
```

Comment:
As discussed on slack, we don't want a new undo level for every failure, only one at the end. So this needs to be changed so that it does something like `editor.undoManager.add()` once the upload of all files is complete.